### PR TITLE
refactor: rename all oss variables to be `--scalar-*`

### PR DESCRIPTION
**Pull request by hwkr**
I'd like to apologize in advance for any of the weird little bugs I may have introduced. Doing a change like this is a little wild.

Updated now with a migration tool and console warning if you have old console variables.

https://github.com/scalar/scalar/assets/6374090/8258f066-d953-4cd4-885b-7132f2d4561a

### CODE REVIEW

**Review by marclave — CHANGES_REQUESTED** [review]
child attributes is missing styles:

image

**Review by amritk — APPROVED** [review]
Looks fine to my rookie eyes

**Review by marclave — CHANGES_REQUESTED** [review]
code block is missing styles for url

current
image


incorrect
image

**Comment by marclave** [comment]
looks like several fonts are off

this pr
image

vs

main
image

**Comment by marclave** [comment]
api client modal has transparent body at bottom

this pr
image

vs main
image

**Comment by marclave** [comment]
response description font is off

pr
image

vs main
image

**Comment by marclave** [comment]
tag description in classic layout is incorrect

pr
image


vs main
image

**Comment by hwkr** [comment]
Alrighty fixed the issues @marclave found, thank you for finding those 🙏 

You can see the changes here, one was a broken line of CSS that wasn't being applied on main and the other one was a default var that was different from the theme var (which can't happen anymore! yay!)

I wasn't able to replicate the issue with the API client body, might need a hand with that.

Google Chrome-2024-04-09-15-06-19@2x

File: packages/api-reference/src/components/MarkdownRenderer/MarkdownRenderer.vue [hunk 1]
```diff
@@ -113,28 +113,19 @@ onServerPrefetch(async () => await sleep(1))
...
   display: block;
   line-height: 1.45;
 }
 .markdown :deep(b),
 .markdown :deep(strong) {
-  font-weight: var(--theme-bold, var(--default-theme-bold));
+  font-weight: var(--scalar-bold);
 }
 .markdown :deep(p) {
-  font-size: var(
```
Comment (hwkr) [inline packages/api-reference/src/components/MarkdownRenderer/MarkdownRenderer.vue:127]: This font size rule wasn't actually being applied anywhere

**Review by marclave — APPROVED** [review]
:)

**Pull request merged**